### PR TITLE
ETL-3057: fix for file not exist during mapper task

### PR DESCRIPTION
**Pull request by ydailinkedin**
Added checks for file existence in extractor.
Another change is in AvroFsHelper. Added method updateFileSystem, which basically makes the fs creation configurable when proxy is enabled. 
Tested on Nertz.
@sahilTakiar @liyinan926 can you review? Thanks!

### CODE REVIEW

File: gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFileExtractor.java [hunk 1]
```diff
@@ -36,10 +37,12 @@ public AvroFileExtractor(WorkUnitState workUnitState) {
   }
 
   @Override
-  public Iterator<GenericRecord> downloadFile(String file)
-      throws IOException {
+  public Iterator<GenericRecord> downloadFile(String file) throws IOException {
     try {
-      return this.closer.register(((AvroFsHelper) this.fsHelper).getAvroFile(file));
```
Comment (sahilTakiar) [inline gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFileExtractor.java]: I think `Closer.register` can handles `null`s

http://grepcode.com/file/repo1.maven.org/maven2/com.google.guava/guava/18.0/com/google/common/io/Closer.java#Closer.register%28java.io.Closeable%29

File: gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFsHelper.java [hunk 2]
```diff
@@ -62,41 +62,54 @@ public FileSystem getFileSystem() {
     return this.fs;
   }
 
+  private void updateFileSystem(String uri) throws IOException, InterruptedException, URISyntaxException {
```
Comment (sahilTakiar) [inline gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFsHelper.java]: Consider changing the method name from `updateFileSystem` to `createFileSystem`

File: gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFsHelper.java [hunk 3]
```diff
@@ -62,41 +62,54 @@ public FileSystem getFileSystem() {
...
+  public void close() throws FileBasedHelperException {
+    /*
+     * TODO
+     * Removing this for now, FileSystem.get() returns the same object each time it is called within a process
+     * If this method gets called in parallel across tasks it is going to cause problems
+     * Basically, close cannot be called multiple times within a process
+     * http://stackoverflow.com/questions/17421218/multiples-hadoop-filesystem-instances
+     */
+
+    //        try {
```
Comment (liyinan926) [inline gobblin-core/src/main/java/gobblin/source/extractor/hadoop/AvroFsHelper.java]: Please remove those lines as this is exactly what git provides: maintain code history so you won't lose it.

**Comment by ydailinkedin** [comment]
Updated.

**Comment by liyinan926** [comment]
LGTM.

**Comment by sahilTakiar** [comment]
LGTM

**Pull request closed without merging**